### PR TITLE
refactor: move some utils to common, clean up

### PR DESCRIPTION
**Pull request by holic**
- Moves `TableId` and `isDefined`/`isNotNull` from `utils` to `common`
- Defines return values of exported functions in `common` in enables linter for it

### CODE REVIEW

File: examples/minimal/packages/client-phaser/src/layers/network/contractComponents.ts [hunk 1]
```diff
@@ -1,24 +0,0 @@
-/* Autogenerated file. Do not edit manually. */
```
Comment (holic) [inline examples/minimal/packages/client-phaser/src/layers/network/contractComponents.ts:1]: this is an unused file and was likely leftover from copy+pasting phaser things (the phaser template removed this file a while back)

**Comment by holic** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1068** <a href="https://app.graphite.dev/github/pr/latticexyz/mud/1068" target="_blank">Graphite</a>  👈
    * **PR #1069** <a href="https://app.graphite.dev/github/pr/latticexyz/mud/1069" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/latticexyz/mud/1068?utm_source=stack-comment).

File: packages/common/src/TableId.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { Hex, stringToHex, hexToString, sliceHex, concatHex } from "viem";
+
+export class TableId {
```
Comment (alvrs) [inline packages/common/src/TableId.ts:3]: should we rename this to `ResourceSelector`? Or have a base class `ResourceSelector` this inherits from? (See https://github.com/latticexyz/mud/pull/999#discussion_r1222208895)

**Review by alvrs — APPROVED** [review]
comment above is not blocking

File: packages/common/src/TableId.ts (hunk 2 above)
Comment (holic) [reply packages/common/src/TableId.ts:3]: ohh I like that better, will do

Comment (holic) [reply packages/common/src/TableId.ts:3]: hmm, gonna save this for later! in searching the codebase for "tableid", we have a lot of instances and so might wanna rethink the boundaries of this

**Pull request merged**